### PR TITLE
fixes on calendar

### PR DESCRIPTION
**Pull request by Fercas123**
- Fixed mozilla div focus bug https://bugzilla.mozilla.org/show_bug.cgi?id=1069739 and added aria label to calendar for accessibility improvement.
- Prevent event defaults in keyboard navigation.

### CODE REVIEW

File: packages/lab/src/calendar/Calendar.tsx [hunk 1]
```diff
@@ -67,6 +73,7 @@ export const Calendar = forwardRef<HTMLDivElement, CalendarProps>(
         <div
           className={clsx(withBaseName(), className)}
           role="application"
+          aria-label={calendarLabel}
```
Comment (origami-z) [inline packages/lab/src/calendar/Calendar.tsx:76]: This could use some test?

File: packages/lab/src/calendar/internal/useFocusManagement.ts [hunk 2]
```diff
@@ -53,7 +53,9 @@ export function useFocusManagement({ date }: { date: DateValue }) {
         break;
       default:
     }
-
+    if (newDate.compare(date) !== 0) {
+      event.preventDefault();
```
Comment (origami-z) [inline packages/lab/src/calendar/internal/useFocusManagement.ts:57]: Not too sure how to test this.

**Review by origami-z — DISMISSED** [review]

**Review by origami-z — APPROVED** [review]

File: packages/lab/src/__tests__/__e2e__/calendar/Calendar.cy.tsx [hunk 3]
```diff
@@ -38,6 +38,15 @@ describe("GIVEN a Calendar", () => {
   describe("Today's Date", () => {
     it("SHOULD set `aria-current=date` on today's date", () => {
       cy.mount(<Default />);
+      cy.findByRole("application").should("exist");
+      cy.findByRole("application").should(
```
Comment (joshwooding) [inline packages/lab/src/__tests__/__e2e__/calendar/Calendar.cy.tsx:42]: should(have.accesibleName) :)

File: .changeset/rude-ravens-design.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+---
+"@salt-ds/lab": patch
+---
+
+Calendar fixes:
+
+- Fixed mozilla div focus bug https://bugzilla.mozilla.org/show_bug.cgi?id=1069739 and added aria label to calendar for accessibility improvement.
```
Comment (joshwooding) [inline .changeset/rude-ravens-design.md:7]: Should be split into two

**Pull request merged**